conda-store-server/conda_store_server/server/views/api.py

```diff
@@ -339,11 +336,171 @@ async def api_update_namespace(
             raise HTTPException(status_code=404, detail="namespace does not exist")
 
         try:
-            api.update_namespace(db, namespace, metadata, role_mappings)
+            res = func(db, namespace, **kwargs)
         except ValueError as e:
             raise HTTPException(status_code=400, detail=str(e.args[0]))
         db.commit()
-        return {"status": "ok"}
+        return res
+
+
+@router_api.put(
+    "/namespace/{namespace}/metadata",
+    response_model=schema.APIAckResponse,
+)
+async def api_update_namespace_metadata(
+    namespace: str,
+    request: Request,
+    metadata: Dict[str, Any] = None,
+    auth=Depends(dependencies.get_auth),
+    conda_store=Depends(dependencies.get_conda_store),
+):
+    _api_namespace_common(
+        namespace=namespace,
+        request=request,
+        auth=auth,
+        conda_store=conda_store,
+        func=api.update_namespace_metadata,
+        metadata_=metadata,
+    )
+    return {"status": "ok"}
```

Comment:
The original method was split into two: metadata and roles. Chris requested this because the old method was doing too much.